### ISSUE TITLE
Crazy Click Behaviour

### ISSUE THREAD

**Issue by SaltyAimbOtter** [BetonQuest/RPGMenu#8]
When i use this event as a click event, the second event is not run.: `
![grafik](https://user-images.githubusercontent.com/40303883/49388022-6bd77b00-f723-11e8-87dc-a0d4cd8e5ea3.png)

**Spigot version:** 
 git-Paper-456 (MC: 1.13.2)
**RPGMenu version:** 
RPGMenu version 3.0
**BetonQuest version:** 
 BetonQuest version 2.0-preview.13
### Description:

When the event is run from a click the new menu is not opend. When the event is run from console it opens. There are no conditions defined :D
https://youtu.be/XVylN2tE_mw

### How to reproduce:
Just like in the video :D
### Checklist:

- [X] I included all versions (use `/version` command)
- [x] I made sure there aren't duplicates of this report [(use Search)](https://github.com/joblo2213/RPGMenu/issues?utf8=%E2%9C%93&q=is%3Aissue+label%3ABug+)
- [X] I made sure I am using an up-to-date version of RPGMenu
- [X] I made sure the bug/error is not caused by any other plugin

**Comment by Momonche** [comment]
Have you try to use a folder with the same events ?

**Comment by bundabrg** [comment]
What's happening is that you click the item, and it runs the events, then closes the menu after the click like its supposed to.

Add the following to the item to prevent the menu closing itself after a click:
```
close: true
```

**Comment by bundabrg** [comment]
Sorry should be close: false

I blame the weather.

**Comment by bundabrg** [comment]
@joblo2213 is there any reason why events are not run after the auto close or can an event programmatically prevent the close if needed?

**Comment by bundabrg** [comment]
Found the issue and have submitted a PR. For the impatient a preview build is [here](https://github.com/bundabrg/RPGMenu/releases/tag/v3.1-preview.1)

Note this change means it does not matter the state of 'close'. If you close the menu in an event it will respect that and not try to auto-close it itself.

**Comment by SaltyAimbOtter** [comment]
It works now, but i get the following error in the console: https://gist.github.com/SaltyAimbOtter/36e386d0ca6f777e9a6eb1b601f14fc3

**Comment by bundabrg** [comment]
Sorted. I think. Try re-download the release.

**Comment by Punishkey** [comment]
Is it possible that the cause of the slots? starts with 0 and in the error it seems that it looks for the total of 18 holes when there are 17 assigned (18 with the 0)

**Comment by bundabrg** [comment]
I can't reproduce the console error.

**Comment by ParadiddlePride** [comment]
I'm now getting the same error: https://pastebin.com/7md4JLuZ
I've tried a couple of things to try to isolate the problem. One thing I noticed is that it only happens when opening and closing menus, any other event occuring when clicking seems to function normally.

A) Button was set to `close: true`, with an event `menu open menu2`
 - Result: Menu closed instantly, 2nd menu did not open.
B) Button set to `close: false`, with an event `menu open menu2`
 - Result: 2nd menu opened, but produced an error in console
C) Button set to `close: false`, with an event `run ^menu close ^menu open menu2`
 - Result: 2nd menu opened, but produced an error in console
D) Button set to `close: false`, with the following:
`menu_openRecipes: run ^menu close ^folder menu2Open delay:1`
`menuTestOpen: menu open menu2`
 - Result: Menu closed, and the 2nd menu opened one second later. No error in console.

From these results, I can only assume there's something going wrong with the timing of the 2nd menu opening when it should be instantaneous.
